contracts/cw3-fixed-multisig/src/contract.rs

```diff
@@ -383,22 +396,61 @@ fn list_voters(
     limit: Option<u32>,
 ) -> StdResult<VoterListResponse> {
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start = start_after.map(|s| Bound::ExclusiveRaw(s.into()));
-
-    let voters = VOTERS
-        .range(deps.storage, start, None, Order::Ascending)
-        .take(limit)
-        .map(|item| {
-            item.map(|(addr, weight)| VoterDetail {
-                addr: addr.into(),
-                weight,
-            })
-        })
-        .collect::<StdResult<_>>()?;
+    let start_address = start_after.map(|addr| Addr::unchecked(addr));
+
+    let addresses = match start_address {
+        Some(addr) => VOTER_ADDRESSES.iter_from(deps.storage, &addr)?,
+        None => VOTER_ADDRESSES.iter(deps.storage),
+    };
+
+    let mut voters = vec![];
+    let mut ctr = 0;
+    for addr in addresses {
+        let weight = match VOTERS.get(deps.storage, &addr) {
+            Some(weight) => weight,
+            None => continue,
+        };
+        voters.push(VoterDetail {
+            addr: addr.to_string(),
+            weight,
+        });
+        ctr += 1;
+        if ctr == limit {
+            break;
+        }
+    }
 
     Ok(VoterListResponse { voters })
 }
 
+/// Utils
+
+fn map_proposal(block: &BlockInfo, item: (u64, Proposal)) -> ProposalResponse {
+    let (id, prop) = item;
+    let status = prop.current_status(block);
+    let threshold = prop.threshold.to_response(prop.total_weight);
+    ProposalResponse {
+        id,
+        title: prop.title,
+        description: prop.description,
+        msgs: prop.msgs,
+        status,
+        expires: prop.expires,
+        threshold,
+    }
+}
+
+// @todo do this with macros
+fn get_proposal(store: &dyn Storage, id: &u64) -> Result<Proposal, ContractError> {
+    PROPOSALS.get(store, id).ok_or(ContractError::NotFound {})
+}
+
+fn get_proposal_std(store: &dyn Storage, id: &u64) -> StdResult<Proposal> {
+    PROPOSALS.get(store, id).ok_or(StdError::NotFound {
+        kind: "Proposal".to_string(),
+    })
+}
```

Comment:
no need for both functions. in general if u can get away with returning StdErrors like `NotFound` throughout a function u should just return StdResult instead of wrapping them in ContractErrors

```suggestion
fn get_proposal(store: &dyn Storage, id: &u64) -> StdResult<Proposal> {
    PROPOSALS.get(store, id).ok_or_else(|| StdError::not_found("Proposal"))
}
```